query/README.md

```diff
@@ -0,0 +1,146 @@
+``` java
+Query query = new QueryBuilder(QueryType.COUNT_RESOURCE)
+        .withEventCollection("<event_collection>")
+        .withGroupBy("click-number")
+        .build();
+QueryResult result = queryClient.execute(query, new RelativeTimeframe("this_month"));
+if (result.isGroupResult()) {
+	for (Map.Entry<Group, QueryResult> groupResult : result.getGroupResults().entrySet()) {
+	    Map<String, Object> groupProperies = groupResult.getKey().getProperties();
+	    long groupCount = groupResult.getValue().longValue();
+	    // ... do something with the group properties and the count result
+	}
+}
+```
+Filtering any query via Interval will cause the query response to be an IntervalResult object. An IntervalResult is a type of QueryResult that consist of Map<AbsoluteTimeframe,QueryResult> objects.
+``` java
+Query query = new QueryBuilder(QueryType.COUNT_RESOURCE)
+        .withEventCollection("<event_collection>")
+        .withInterval("weekly")
+        .build();
+QueryResult result = queryClient.execute(query, new RelativeTimeframe("this_year"));
+if (result.isIntervalResult()) {
+        for (Map.Entry<AbsoluteTimeframe, QueryResult> intervalResult : result.getIntervalResults().entrySet()) {
+            AbsoluteTimeframe timeframe = intervalResult.getKey();
+            long intervalCount = intervalResult.getValue().longValue();
+            // ... do something with the absolute timeframe and count result.
+        }
+}      
+```
+Filtering via both Group By and Interval will cause the query response to be an IntervalResult object that contains GroupByResult objects follows:
+
+``` java
+Query query = new QueryBuilder(QueryType.COUNT_RESOURCE)
+        .withEventCollection("<event_collection>")
+        .withInterval("weekly")
+        .withGroupBy("click-number")
+        .build();
+QueryResult result = queryClient.execute(query, new RelativeTimeframe("this_year"));
+
+if (result.isIntervalResult()) {
+    for (Map.Entry<AbsoluteTimeframe, QueryResult> intervalResult : result.getIntervalResults().entrySet()) {
+        AbsoluteTimeframe timeframe = intervalResult.getKey();
+
+        for (Map.Entry<Group, QueryResult> groupResult : intervalResult.getValue().getGroupResults().entrySet()) {
+            Map<String, Object> groupProperies = groupResult.getKey().getProperties();
+            long groupCount = groupResult.getValue().longValue();
+            // ... do something with the group properties and the count result
+        }
+    }
+}
+```
+
+### Utility Methods
+
+There are also some utility methods to add filters and absolute timeframes to a Query:
+```java
+
+// notice this will add two filter parameters, with 1 < click-count < 5
+Query query = new QueryBuilder(QueryType.COUNT_RESOURCE)
+	            .withEventCollection("<event_collection>")
+	            .withFilter("click-count", "lt", 5)
```

Comment:
Concern: It feels a little bit strange to me that you specify the `QueryType` with an enum but you specify the filter operator with a raw string. They're both basically a value from a fixed, relatively short list, so I think people would naturally expect them to be handled in the same way. I think either way is probably fine but I feel like we should be consistent. Thoughts?